Fix crash on clicking classes/search or editor/project settings

chongdashu opened the pull request:
As a result of a bugfix for menu items not releasing the "selected" state
after initial mouse clicks (#14854), the logic was inadvertently causing
gui mouse events to recurse infinitely as a result of a multi-level call.

The solution is to switch it to a deferred call for the mouse input.

This fixes #14886.

akien-mga commented:
Could you squash the two commits together? The cleanup is simple enough to go with the first commit.

chongdashu commented:
Will do so!

chongdashu committed:
Fix crash on editor settings and help/classes menu

As a result of a bugfix for menu items not releasing the "selected" state
after initial mouse clicks (#14854), the logic was inadvertently causing
gui mouse events to recurse infinitely as a result of a multi-level call.

The solution is to switch it to a deferred call for the mouse input.

This fixes #14886.

chongdashu commented:
@akien-mga -- thanks for the advice, the commits have been squashed now.

volzhs commented:
superseded by 1fa9aac?

reduz commented:
seems like i fixed this a bit ago
